### PR TITLE
init_commit review suggestions

### PR DESCRIPTION
**Pull request by nandubatchu**
addressing review comments from https://github.com/govind1246/js-sdk/pull/1

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -109,13 +108,12 @@ export class PayIDClaim implements ICruxPayClaim {
...
-            const encryptionKeyDigest = await this._encryption.digest(encryptionKey);
-            this.identitySecrets = (await this._encryption.decryptJSON(encryptedObj.encBuffer, encryptedObj.iv, encryptionKeyDigest) as nameService.IIdentityClaim);
+            this.identitySecrets = (await this._encryption.decryptJSON(encryptedObj.encBuffer, encryptedObj.iv, encryptionKey) as nameService.IIdentityClaim);
         }
     }
 
-    public toJSON = (): ICruxPayClaim => {
-        // await this.encrypt()
+    public toJSON = async (): Promise<ICruxPayClaim> => {
+        await this.encrypt();
```
Comment (quixote911) [inline src/index.ts]: Makes more sense for this to be a part of the `save` method.

File: src/packages/name-service/utils.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+import config from "../../config";
 import { ErrorHelper, PackageErrorCode } from "../error";
 import { cachedFunctionCall, httpJSONRequest } from "../utils";
 
 const log = getLogger(__filename);
 
-export const fetchNameDetails = async (blockstackId: string, bnsNodes: string[]): Promise<object|undefined> => {
-    const nodeResponses = bnsNodes.map((baseUrl) => bnsResolveName(baseUrl, blockstackId));
+export const fetchNameDetails = async (blockstackId: string, bnsNodes?: string[]): Promise<object|undefined> => {
+    bnsNodes = bnsNodes || config.BLOCKSTACK.BNS_NODES;
```
Comment (quixote911) [inline src/packages/name-service/utils.ts]: Have this be a mandatory param, utils shouldn't have anything to do with config.

File: src/packages/configuration-service.ts [hunk 3]
```diff
@@ -71,10 +71,8 @@ export class BlockstackConfigurationService extends NameServiceConfigurationServ
         let ns: nameservice.BlockstackService;
         let gaiaHub: string | undefined;
         if (this.blockstackID) {
-            const gaiaUrls = await getGaiaDataFromBlockstackID(this.blockstackID);
-            if (gaiaUrls.gaiaWriteUrl) {
-                gaiaHub = gaiaUrls.gaiaWriteUrl;
-            }
+            const gaiaUrls = await getGaiaDataFromBlockstackID(this.blockstackID, (this.clientConfig.nameserviceConfiguration && this.clientConfig.nameserviceConfiguration.bnsNodes) || config.BLOCKSTACK.BNS_NODES);
+            gaiaHub = gaiaUrls.gaiaWriteUrl || undefined;
```
Comment (SanchayPahalwani) [inline src/packages/configuration-service.ts]: Do we need "|| undefined" here?

File: src/packages/name-service/utils.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { AssertionError, deepStrictEqual } from "assert";
 import { getLogger } from "../..";
+import config from "../../config";
```
Comment (SanchayPahalwani) [inline src/packages/name-service/utils.ts]: Remove unused import

File: src/test/cruxclient.ts [hunk 5]
```diff
@@ -33,7 +33,7 @@ describe('CruxClient tests', () => {
...
+		'payIDClaim': {"virtualAddress":"syedhassanashraf@cruxdev.crux","identitySecrets":"{\"iv\":\"d6mQLg2lv/SHvpPV\",\"encBuffer\":\"EJ0NDIGyvT+FMxjG9pD7yKkUVUJffcd0mU3V7rqgbpip5ikCR+kx/Xl4tJcBNtQUJzKbDPNJeeWdzUKH01NQOiyEe9qfCYEqzbJgj4PSqE/B1geSlCMRWcdJMdbrUDHgEcwDWw6PC41p0odNHLzMWOR1LA+PKVzCAMXRom6UppG+gjMnjp9QqF00SfnNR6wla/es7LgE3eF6O4Gq9Ku/mAhdYrZiaMMd47hQvTzH9KzDZrTjK37jmtiRC6cuNDy9zZ+BtpzufozAbuCvfeQc2nXKdV4a3kx7peLK7eAGZUV6soaOfq+ZIPx5bbnNos8Py7fhNOmqbop12yCQ4Ot2jm7Bmx9eXIFq7/EtLWP488xU3l2qB0/XHMmGtlsZ8Er19al+aB4OJAJ5yzBU01rVQFVkOLC50ZlYT4VcVgJbOUvRXi8c4mlv0JNfvajA\"}"},
 		'cruxID': 'syedhassanashraf@cruxdev.crux',
 		'blockStackID': 'syedhassanashraf.crux.id',
```
Comment (SanchayPahalwani) [inline src/test/cruxclient.ts]: Remove old BlockstackID as it is not being used in any test cases, also that's why went undetected.

**Pull request merged**